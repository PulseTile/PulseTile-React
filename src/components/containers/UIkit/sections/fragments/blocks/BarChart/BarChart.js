import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { lifecycle } from 'recompose';

import PatientsChart from '../../../../../../containers/PatientsChart/PatientsChart';
import { patientsDepartments } from '../../../../../../../config/patients.constants';
import patientsSelector from './selectors';
import { fetchPatientsRequest } from '../../../../../../../ducks/feth-patients.duck';
import { fetchPatientsOnMount } from '../../../../../../../utils/HOCs/fetch-patients.utils';

const mapDispatchToProps = dispatch => ({ actions: bindActionCreators({ fetchPatientsRequest }, dispatch) });

@connect(patientsSelector, mapDispatchToProps)
@lifecycle(fetchPatientsOnMount)

/**
 * This component returns content of BarChart section in Blocks
 */
export default class BarChart extends Component {

    render() {
        const { patientsByDepartment } = this.props;
        return (
            <div id="charts" className="ui-section">
                <strong className="ui-title">Charts</strong>
                <div className="ui-sub-section">
                    <strong className="ui-sub-title">Chart Bar</strong>
                    <div className="panel panel-secondary">
                        <div className="panel-heading">
                            <h3 className="panel-title"><i className="fa fa-bar-chart"></i> Patients By Setting</h3>
                        </div>
                        <div className="panel-body">
                            <div className="panel-body-inner">
                                <div className="chart-block">
                                    <PatientsChart
                                        title='Patients By Setting'
                                        subTitle='This is a brief description of patients by setting.'
                                        borderColor="rgba(36, 161, 116,1)"
                                        backgroundColor="rgba(36, 161, 116,0.3)"
                                        isChartsDataReceived
                                        labels={patientsDepartments}
                                        patients={patientsByDepartment}
                                        onBarClick={function () {}}
                                    />
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
};
