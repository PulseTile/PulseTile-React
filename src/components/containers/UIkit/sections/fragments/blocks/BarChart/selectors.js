import _ from 'lodash/fp';
import { createSelector } from 'reselect';
import { patientsDepartments } from '../../../../../../../config/patients.constants';

export const patientsByDepartmentsSelector = ({ patients }) => patientsDepartments.map(department => _.filter(department.predicate)(patients));

const patientsSelector = createSelector(
    patientsByDepartmentsSelector,
    (patientsByDepartment => {
        return {
            patientsByDepartment: patientsByDepartment
        }
    })
);

export default patientsSelector;
