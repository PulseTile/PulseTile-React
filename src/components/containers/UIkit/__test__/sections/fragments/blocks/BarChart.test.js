import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import BarChart from '../../../../sections/fragments/blocks/BarChart/BarChart';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();
const store = mockStore({
    isSidebarVisible: true,
    isTouchDevice: true,
});

const patientsByDepartment = [];

describe('Component <BarChart />', () => {
  it('should renders BarChart with props correctly', () => {
    const component = shallow(<BarChart patientsByDepartment={patientsByDepartment} store={store} />).dive();
    expect(component).toMatchSnapshot();
  });
});
