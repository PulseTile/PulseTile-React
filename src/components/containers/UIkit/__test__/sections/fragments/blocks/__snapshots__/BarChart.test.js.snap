// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <BarChart /> should renders BarChart with props correctly 1`] = `
<BarChart
  actions={
    Object {
      "fetchPatientsRequest": [Function],
    }
  }
  patientsByDepartment={
    Array [
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
